sidekick_core/lib/src/update/migration.dart

```diff
@@ -55,6 +63,57 @@ class _InlineMigrationStep extends MigrationStep {
   }
 }
 
+class GitPatchMigrationStep extends MigrationStep {
+  GitPatchMigrationStep(
+    this.patch, {
+    required this.description,
+    this.pullRequestLink,
+    required Version targetVersion,
+  }) : super(
+          name:
+              '$description${pullRequestLink != null ? ' ($pullRequestLink)' : ''}',
+          targetVersion: targetVersion,
+        );
+
+  /// The git patch to be applied for this migration step
+  final String patch;
```

Comment:
It might be more future-proof to have a `String Function(MigrationContext)`. I don't think we can always access all information statically in to future

Reply:
Changed it here 8fcacedab7872e05343dc2286a175733ab6791ec

We can in the future inject some `Context` that gives us access to the file location